includes/server/class-llms-rest-access-plans-controller.php

```diff
@@ -0,0 +1,1081 @@
+
+			$access_plan = $this->get_object( (int) $request['id'] );
+
+			if ( is_wp_error( $access_plan ) ) {
+				return $can_update;
+			}
+
+			$product_id               = $access_plan->get( 'product_id' );
+			$product_post_type_object = get_post_type_object( get_post_type( $product_id ) );
+
+			if ( ! current_user_can( $product_post_type_object->cap->edit_post, $product_id ) ) {
+				return llms_rest_authorization_required_error(
+					sprintf(
+						// Translators: %s = The post type name.
+						__( 'Sorry, you are not allowed to update %s as this user.', 'lifterlms' ),
+						$access_plan->get_post_type_label( 'name' )
+					)
+				);
+			}
+
+			$can_update = true;
+		}
+
+		return $can_update;
+	}
+
+	/**
+	 * Check if a given request has access to delete an item
+	 *
+	 * @since [version]
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error
+	 */
+	public function delete_item_permissions_check( $request ) {
+
+		$can_delete = parent::delete_item_permissions_check( $request );
+
+		// If current user cannot delete the item because of authorization, check if the current user can edit the "parent" course/membership.
+		if ( llms_rest_is_authorization_required_error( $can_delete ) ) {
+
+			$access_plan = $this->get_object( (int) $request['id'] );
+
+			if ( is_wp_error( $access_plan ) ) {
+				return $can_delete;
+			}
+
+			$product_id               = $access_plan->get( 'product_id' );
+			$product_post_type_object = get_post_type_object( get_post_type( $product_id ) );
+
+			if ( ! current_user_can( $product_post_type_object->cap->edit_post, $product_id ) ) {
+				return llms_rest_authorization_required_error(
+					sprintf(
+						// Translators: %s = The post type name.
+						__( 'Sorry, you are not allowed to delete %s as this user.', 'lifterlms' ),
+						$access_plan->get_post_type_label( 'name' )
+					)
+				);
+			}
+
+			$can_delete = true;
+		}
+
+		return $can_delete;
+	}
+
+	/**
+	 * Prepare links for the request
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Access_Plan $access_plan LLMS Access Plan instance.
+	 * @param WP_REST_Request  $request     Request object.
+	 * @return array Links for the given object.
+	 */
+	protected function prepare_links( $access_plan, $request ) {
+
+		$links = parent::prepare_links( $access_plan, $request );
+		unset( $links['content'] );
+
+		$id         = $access_plan->get( 'id' );
+		$product_id = $access_plan->get( 'product_id' );
+		$post       = llms_get_post( $product_id );
+
+		// Post: the check below should be redundant but at the moment we don't delete
+		// access plans when deleting "related" courses/memberships.
+		if ( $post instanceof LLMS_Course || $post instanceof LLMS_Membership ) {
+			$links['post'] = array(
+				'href' => rest_url(
+					sprintf(
+						'%s/%s/%s',
+						'llms/v1',
+						$access_plan->get_product_type(),
+						$product_id
+					)
+				),
+			);
+		}
```

Comment:
I don't like having a hack in here for a bug that we can fix in the core... I think we should remove this and just fix the core bug simultaneously: https://github.com/gocodebox/lifterlms/issues/263

Reply:
That said... I'm okay with shipping it if we open a new issue (in rest) so that we can make sure we don't forget about it when we update the core

Reply:
Yeah but unless we also update the db removing orphans access plans, the check here is needed.
So yeah, let's also update the db.

Reply:
okay sounds good

Reply:
I'm on this right now (db update).
Did the ap deletion on course/membership deletion.

Reply:
ap clen up in core, see https://github.com/gocodebox/lifterlms/pull/1518